src/vs/editor/contrib/quickFix/quickFix.ts

```diff
@@ -31,7 +31,55 @@ export function getCodeActions(model: IReadOnlyModel, range: Range): TPromise<Co
 		});
 	});
 
-	return TPromise.join(promises).then(() => allResults);
+	return TPromise.join(promises).then(
+		() => allResults.sort(codeActionsComparator)
+	);
+}
+
+function codeActionsComparator(a: CodeAction, b: CodeAction): number {
+
+	if (a.command) {
+		if (b.command) {
+			return a.title.localeCompare(b.title);
+		} else {
+			return 1;
+		}
+	}
+	else if (b.command) {
+		return -1;
+	}
+	else {
+		return suggestionsComparator(a, b);
+	}
```

Comment:
Nice, looks a lot better. One or two small things:

* don't look at the command, only at diagnostics
* don't compare on CodeAction#title but only on the diagnostic. So, compare diagnostics-messages when both have diagnostics, if only one code action has diagnostics it "wins", otherwise they are considered equal.